promising/promise.py

```
@@ -164,6 +163,103 @@ def __init__(
+        """
+        try:
+            running_loop = asyncio.get_running_loop()
+        except RuntimeError:
+            running_loop = None
+
+        if running_loop is self._loop:
+            raise PromiseError(
+                "promise.sync() cannot be called from the "
+                "event loop thread because it would deadlock. "
+                "Use 'await promise' instead."
+            )
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)